File: app/src/main/java/com/vocaby/app/utils/Logger.java
package com.vocaby.app.utils;
import android.util.Log;

import com.bugsnag.android.Bugsnag;
import com.vocaby.app.Constants;

public class Logger {
    public static void reportErrorToBugsnag(Throwable error) {
        Bugsnag.notify(error);
    }
    public static void reportErrorToDebug(Throwable error) {
        Log.d(Constants.DEBUG_TAG, error.getMessage());
    }
    public static void reportToDebug(String message) {
        Log.d(Constants.DEBUG_TAG, message);
    }
}

File: app/src/main/java/com/vocaby/app/utils/Logger.kt
package com.vocaby.app.utils

import android.util.Log
import com.bugsnag.android.Bugsnag
import com.vocaby.app.Constants

object Logger {
    fun reportErrorToDebug(error: Throwable) {
        error.message?.let { message ->
            Log.d(Constants.DEBUG_TAG, message)
            Log.d(Constants.DEBUG_TAG, error.stackTraceToString())
        } ?: Log.d(Constants.DEBUG_TAG, "There was an error..." + error.javaClass)
    }

    fun reportErrorToBugsnag(error: Throwable) {
        Bugsnag.notify(error)
    }

    fun reportToDebug(message: String?) {
        message?.let {
            Log.d(Constants.DEBUG_TAG, message)
        }
    }
}